base/loading.jl

```diff
@@ -91,6 +91,17 @@ struct SHA1
     end
 end
 SHA1(s::AbstractString) = SHA1(hex2bytes(s))
+parse(::Type{SHA1}, s::AbstractString) = SHA1(s)
+function tryparse(::Type{SHA1}, s::AbstractString)
+    try
+        return parse(SHA1, s)
+    catch e
+        if isa(e, InterruptException)
```

Comment:
Wouldn't it be better to specifically return `nothing` for an `ArgumentError` and `rethrow` everything else?

Reply:
Yes, that does seem better to me.